internal/startup/auto_operator_configuration_handler.go

```diff
@@ -70,22 +80,38 @@ func (r *AutoOperatorConfigurationResourceHandler) CreateOrUpdateOperatorConfigu
 
 	operatorConfigurationResource := convertValuesToResource(operatorConfigurationValues)
 	go func() {
+		// If multiple replicas are active, only the leader should attempt to create or update the operator
+		// configuration resource.
+		logger.Info(
+			"waiting for this replica to become leader before creating or updating the Dash0 operator configuration " +
+				"resource",
+		)
+		// Block until NotifiyOperatorManagerJustBecameLeader has been called.
+		select {
+		case <-ctx.Done():
+			logger.Error(ctx.Err(), "context cancelled while waiting for this replica to become leader")
+			return
+		case <-r.hasBecomeLeaderChan:
+			logger.Info("this replica has become leader, proceeding with creating/updating the operator configuration resource")
+		}
```

Comment:
^ this is also only tangentially related to the new readiness check -- previously, all replicas would try to create/update the operator configuration resource, now this depends on leader election, so only one pod tries to do that.